src/openforms/translations/tests/test_api.py

```diff
@@ -0,0 +1,131 @@
+from collections import namedtuple
+
+from django.test import override_settings
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from hypothesis import given, settings, strategies as st
+
+# from hypothesis.extra.django import TestCase
+from rest_framework.test import APISimpleTestCase
+
+Language = namedtuple("Language", "conf local")
+
+
+class I18NAPITests(APISimpleTestCase):
+    databases = ["default"]
+
+    @given(
+        st.lists(
+            st.sampled_from(
+                [
+                    Language(("en", _("English")), "English"),
+                    Language(("nl", _("Dutch")), "Nederlands"),
+                    Language(("tr", _("Turkish")), "Türkçe"),
+                ]
+            ),
+            unique=True,
+        )
+    )
+    @settings(deadline=500)
+    def test_info_contains_available_langauges(self, languages):
+        with override_settings(
+            USE_I18N=True,
+            LANGUAGES=[lang.conf for lang in languages],
+        ):
+            # expect language names in their local representation
+            expected = [
+                {"code": lang.conf[0], "name": lang.local} for lang in languages
+            ]
+            url = reverse("api:i18n:info")
+            response = self.client.get(url)
+            available = response.json()["langauges"]
+            self.assertEqual(available, expected)
+
+    @override_settings(
+        USE_I18N=True,
+        LANGUAGE_CODE="en",
+        LANGUAGES=[
+            ("en", _("English")),
+            ("nl", _("Dutch")),
+        ],
+    )
+    def test_info_contains_active_language(self):
+        # default fallback
+        url = reverse("api:i18n:info")
+        response = self.client.get(url)
+        current = response.json()["current"]
+        self.assertEqual(current, "en")
+
+    @override_settings(
+        USE_I18N=True,
+        LANGUAGE_CODE="en",
+        LANGUAGES=[
+            ("en", _("English")),
+            ("nl", _("Dutch")),
+        ],
+        MIDDLEWARE=[
+            "django.contrib.sessions.middleware.SessionMiddleware",
+            "django.middleware.locale.LocaleMiddleware",
+            "django.middleware.common.CommonMiddleware",
+        ],
```

Comment:
you can leave this out to test our settings/configuration isn't missing a/the middleware (Steven's PR got merged that enabled the middleware, so you can rebase this)